### PR TITLE
Hotfix/random proposals until valid evals

### PR DESCRIPTION
**Pull request by LGro**
If `num_initial_random_samples` is specified > 0, keep sampling random specifications until `num_initial_random_samples`  many *valid* evaluations were reported.

### CODE REVIEW

File: tests/optimizers/botorch_base_test.py [hunk 1]
```diff
@@ -166,6 +166,56 @@ def test_find_optimum_in_1d_discrete_space(seed):
...
+        es.create_evaluation(objectives={"loss": None}),
+    )
+
+    es = opt.generate_evaluation_specification()
+    assert not es.optimizer_info[
+        "model_based_pick"
+    ], "One valid evaluation reported, 1 < 2 initial random samples"
+    opt.report(
+        es.create_evaluation(objectives={"loss": es.configuration["integ"] ** 2}),
+    )
```
Comment (LGro) [inline tests/optimizers/botorch_base_test.py:211]: Before the PR, this would have already been a model based pick.

File: tests/optimizers/botorch_base_test.py [hunk 2]
```diff
@@ -166,6 +166,56 @@ def test_find_optimum_in_1d_discrete_space(seed):
...
+        "model_based_pick"
+    ], "One evaluation reported, 1 < 2 initial random samples"
+    opt.report(
+        es.create_evaluation(objectives={"loss": None}),
+    )
+
+    es = opt.generate_evaluation_specification()
+    assert not es.optimizer_info[
+        "model_based_pick"
+    ], "One valid evaluation reported, 1 < 2 initial random samples"
```
Comment (LGro) [inline tests/optimizers/botorch_base_test.py:208]: Before this PR, this would have already been a model based pick.

File: blackboxopt/optimizers/botorch_base.py [hunk 3]
```diff
@@ -277,15 +280,20 @@ def generate_evaluation_specification(self) -> EvaluationSpecification:
 
         if self.num_initial_random > 0 and (
             self.X.size(-2) < self.num_initial_random
-            or torch.nonzero(~torch.any(self.losses.isnan(), dim=1)).numel() == 0
+            or (
+                torch.nonzero(~torch.any(self.losses.isnan(), dim=1)).numel()
+                < self.num_initial_random
+            )
```
Comment (sfalkner) [inline blackboxopt/optimizers/botorch_base.py:286]: Note, that the comment below now is wrong, because sampling random configurations now continues until `num_initial_random` many have all valid objectives.

File: blackboxopt/optimizers/botorch_base.py [hunk 4]
```diff
@@ -277,15 +280,20 @@ def generate_evaluation_specification(self) -> EvaluationSpecification:
 
         if self.num_initial_random > 0 and (
             self.X.size(-2) < self.num_initial_random
```
Comment (sfalkner) [inline blackboxopt/optimizers/botorch_base.py:282]: I believe this line is redundant. Both `self.losses` and `self.X` are always of the same size and this line is always satisfied, if the condition below is satisfied. Or am I missing a corner case here?

File: tests/optimizers/botorch_base_test.py (hunk 2 above)
Comment (sfalkner) [reply tests/optimizers/botorch_base_test.py:208]: True, but it was consistent with the behavior described in the in-line comment.
But I like this solution now much better.

**Review by sfalkner — APPROVED** [review]
Thanks @LGro for making this more consistent.

File: blackboxopt/optimizers/botorch_base.py (hunk 4 above)
Comment (LGro) [reply blackboxopt/optimizers/botorch_base.py:282]: You're right, I'll simplify accordingly.
